### DISCUSSION TITLE
Error while trying to build the mod "Task :fabric:compileJava FAILED"

### DISCUSSION THREAD

**Discussion by CaptainRexPL** [FabricMC/fabric-api#2523] [Mod Dev Support]
When I try to build a mod I get this error:
"gradle build

> Configure project :fabric
Fabric Loom: 1.0.7
Loom is applying dependency data manually to publications instead of using a software component (from(components["java"])). This is deprecated.

> Task :fabric:compileJava FAILED
/home/captainrexpl/IdeaProjects/api-mod/fabric/src/main/java/com/collarmc/fabric/mixins/ClientPlayNetworkHandlerMixin.java:19: error: cannot find symbol
        PlayerEntity player = MinecraftClient.getInstance().world.getPlayerByUuid(packet.getSender());
                                                                                        ^
  symbol:   method getSender()
  location: variable packet of type GameMessageS2CPacket
/home/captainrexpl/IdeaProjects/api-mod/fabric/src/main/java/com/collarmc/fabric/mixins/ClientPlayNetworkHandlerMixin.java:20: error: cannot find symbol
        if (CollarClient.getChatService().onChatMessageReceived(player, packet.getMessage(), packet.getlocation())) {
                                                                              ^
  symbol:   method getMessage()
  location: variable packet of type GameMessageS2CPacket
/home/captainrexpl/IdeaProjects/api-mod/fabric/src/main/java/com/collarmc/fabric/mixins/ClientPlayNetworkHandlerMixin.java:20: error: cannot find symbol
        if (CollarClient.getChatService().onChatMessageReceived(player, packet.getMessage(), packet.getlocation())) {
                                                                                                   ^
  symbol:   method getlocation()
  location: variable packet of type GameMessageS2CPacket
/home/captainrexpl/IdeaProjects/api-mod/fabric/src/main/java/com/collarmc/fabric/client/messaging/GroupChatInterceptor.java:43: error: incomparable types: MessageType and RegistryKey<MessageType>
        if (location != MessageType.CHAT) {
                     ^
4 errors
Deprecated Loom APIs were used in this build, making it incompatible with future versions of Loom. Use Gradle warning modes to control the verbosity of the warnings.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':fabric:compileJava'.
> Compilation failed; see the compiler error output for details.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 3s
1 actionable task: 1 executed"

yarn mappings version: 1.19.2+build.5

The mod was originally made for 1.17.

**Comment by warjort** [comment]
Looks to me like you need to update your mod for the changes made in chat in 1.19
e.g. https://www.minecraft.net/en-us/article/minecraft-java-edition-1-19-1

**Reply by CaptainRexPL** [reply]
I read this article, but there's nothing that could cause this error with e.g. getsender(). I checked the directory where "getsender", "getlocation" etc. originally were and it looks like someone got rid of it. I also checked if this was caused by chat changes in 1.19.1/1.19.2 so I tried to downgrade the version to 1.19 but the error is still here.

**Reply by LogOnACouch** [reply]
This problem happened to me too, but i am on fabric 1.20.1. It won't build.
